WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanViewModel.kt

```diff
@@ -1,15 +1,65 @@
 package org.wordpress.android.ui.jetpack.scan
 
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModel
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.scan.ScanStateModel
+import org.wordpress.android.fluxc.model.scan.ScanStateModel.State
+import org.wordpress.android.ui.jetpack.scan.ScanListItemState.ScanState.ScanIdleState
+import org.wordpress.android.ui.jetpack.scan.ScanListItemState.ScanState.ScanScanningState
+import org.wordpress.android.ui.jetpack.scan.ScanViewModel.UiState.Content
 import javax.inject.Inject
 
-class ScanViewModel @Inject constructor() : ViewModel() {
+class ScanViewModel @Inject constructor(
+    private val scanStatusService: ScanStatusService
+) : ViewModel() {
     private var isStarted = false
+    private var lastScanState: State? = null
 
-    fun start() {
+    private val _uiState: MutableLiveData<UiState> = MutableLiveData()
+    val uiState: LiveData<UiState> = _uiState
+
+    private val scanStateObserver = Observer<ScanStateModel> {
+        if (it.state != lastScanState) {
+            lastScanState = it.state
+            _uiState.value = buildContentUiState(it)
+        }
+    }
+
+    lateinit var site: SiteModel
+
+    fun start(site: SiteModel) {
         if (isStarted) {
             return
         }
+        this.site = site
+        init()
         isStarted = true
     }
+
+    private fun init() {
+        scanStatusService.scanState.observeForever(scanStateObserver)
+        scanStatusService.start(site)
+    }
+
+    private fun buildContentUiState(model: ScanStateModel): Content {
+        val scanStateItem = when (model.state) {
+            State.IDLE -> model.threats?.let { ScanIdleState.ThreatsFound() } ?: ScanIdleState.ThreatsNotFound()
+            State.SCANNING -> ScanScanningState()
+            State.PROVISIONING, State.UNAVAILABLE, State.UNKNOWN -> ScanScanningState() // TODO: ashiagr filter out
```

Comment:
Idea (💡 ): I usually suggest every `when` case to stand by itself, even if the action is duplicated. I believe this improves the readability and makes sure that creepy bugs are avoided. Consider the below instead, but again, this is very subjective:
```
        val scanStateItem = when (model.state) {
            ScanStateModel.State.IDLE -> {
                model.threats?.let { ScanIdleState.ThreatsFound() } ?: ScanIdleState.ThreatsNotFound()
            }
            ScanStateModel.State.SCANNING -> ScanScanningState()
            ScanStateModel.State.PROVISIONING -> ScanScanningState()
            ScanStateModel.State.UNAVAILABLE -> ScanScanningState()
            ScanStateModel.State.UNKNOWN -> ScanScanningState()
        }
```

PS: I can see that `ScanScanningState()` is used on all four cases, but you split it in two, why is that?

Reply:
> I can see that ScanScanningState() is used on all four cases, but you split it in two, why is that?

For `SCANNING` I was sure we can use `ScanScanningState`(). I clubbed the rest as I was undecided if we should use that state (ideally should not as those states don't support scanning).